TACC/Core-Portal
# task/FP-1079: Catch when a system is down for maintenance and alert user

## Overview: ##
Compared data files storage system or app's executing system against system monitor to determine if a 'system maintenance' message needs to be displayed.

## Related Jira tickets: ##

* [FP-1079](https://jira.tacc.utexas.edu/browse/FP-1079)

## Summary of Changes: ##
- Added an array to both DataFilesTable and AppForm which grab systems which are down
- Added a var to both DataFilesTable and AppForm which grabs the relevant storage/execution system
- Added a 'system maintenance' message which displays should the current system var be in the down systems array

## Testing Steps: ##
For DataFilesTable (file changes in DataFilesTable.js):
1. Line 33 becomes: `const err = useSelector(state => state.files.error[section]='502');`
2. Add @ line 42: `useSelector(state => state.systemMonitor.list.map(sys => sys.is_operational = false));`
3. Add @ line 275: `data = [];`
4. Navigate to any of the 'My Data' storage systems in the Data Files tab and confirm a 'system maintenance' message appears.

For AppForm:
1. Add @ line 170 in AppForm.js: `state.systemMonitor.list.map(sys => sys.is_operational = false);`
2. Navigate to one of the apps and confirm a 'system maintenance' message appears.

## UI Photos:
Data Files:

![image](https://user-images.githubusercontent.com/43251554/137805069-9b65e2ec-c407-45ac-b3f0-bca8d65be654.png)

App Form:

![image](https://user-images.githubusercontent.com/43251554/137804503-5294ecb8-12b1-49c9-8b27-3620cbb2671a.png)


## Notes: ##
